include/mbedtls/net_sockets.h

```diff
@@ -283,16 +283,16 @@ int mbedtls_net_recv_timeout(void *ctx, unsigned char *buf, size_t len,
                              uint32_t timeout);
 
 /**
- * \brief          Closes down the connection and free associated data
+ * \brief          Close down the connection and clear the context
```

Comment:
It's good to clarify that our `free` functions have a misleading name. But in this case I don't think the reword is really an improvement. The old wording said “free associated data”, not freeing the context object itself, so it was less misleading than most. The new wording doesn't make it clear that this frees all resources associated with the connection.

Here I think it would be clearer to keep the old wording, and add a sentence (not in the brief description) saying that after calling this function, you can reuse the context object, and if you don't want to do that you should free the context object itself if it was dynamically allocated.

Reply:
I've filed https://github.com/Mbed-TLS/mbedtls/issues/7837 for the general issue. We've been talking about it since at least 2018 but never found the time to act.